### PR TITLE
ci: update the minimum kubernetes version supported

### LINKED ISSUE #12365: For v1.12 update the minimum K8s version supported to v1.22
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
For v1.12 update the minimum K8s version supported to v1.22, to continue running CI against the most recent six K8s versions for the latest release.

A similar change is needed compared to #11436. 

For the CI where four K8s versions are run, let's use:
`["v1.22.17", "v1.24.14", "v1.26.5", "v1.27.2"]` 

For the CI where four K8s versions are run, let's use:
`["v1.22.17", "v1.27.2"]` 

**What is use case behind this feature:**
Follow the [K8s version support policy](https://rook.io/docs/rook/latest/Getting-Started/release-cycle/#k8s-versions)

### PR DESCRIPTION
**Pull request by Sheetalpamecha**
**Description of your changes:**
For v1.12 update the minimum K8s version supported to v1.22, to continue running CI against the most recent six K8s versions for the latest release.

**Which issue is resolved by this Pull Request:**
Resolves #12365 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by travisn — CHANGES_REQUESTED** [review]
The changes look good, please just add a brief commit description paragraph as mentioned in the [commit structure](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure), thanks!

File: Documentation/Getting-Started/Prerequisites/prerequisites.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ and Rook is granted the required privileges (see below for more information).
 
 ## Minimum Version
 
-Kubernetes **v1.21** or higher is supported.
+Kubernetes **v1.22** or higher is supported.
```
Comment (subhamkrai) [inline Documentation/Getting-Started/Prerequisites/prerequisites.md:10]:
```suggestion
Kubernetes **v1.22.17** or higher is supported.
```

can we be more specific?

File: .github/workflows/canary-test-config/action.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ runs:
     - name: Setup Minikube
       shell: bash --noprofile --norc -eo pipefail -x {0}
       run: |
-        tests/scripts/github-action-helper.sh install_minikube_with_none_driver v1.27.1
+        tests/scripts/github-action-helper.sh install_minikube_with_none_driver v1.27.2
```
Comment (subhamkrai) [inline .github/workflows/canary-test-config/action.yaml:19]:
```suggestion
        tests/scripts/github-action-helper.sh install_minikube_with_none_driver v1.27.3
```
v1.27.3 is the latest Kubernetes release, let's move to that.

**Review by subhamkrai — CHANGES_REQUESTED** [review]

File: Documentation/Getting-Started/Prerequisites/prerequisites.md (hunk 1 above)
Comment (Sheetalpamecha) [reply Documentation/Getting-Started/Prerequisites/prerequisites.md:10]: Sure, changed at other places in Doc too.

File: .github/workflows/canary-test-config/action.yaml (hunk 2 above)
Comment (Sheetalpamecha) [reply .github/workflows/canary-test-config/action.yaml:19]: Done.

**Review by subhamkrai — APPROVED** [review]

File: Documentation/Getting-Started/Prerequisites/prerequisites.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ and Rook is granted the required privileges (see below for more information).
 
 ## Minimum Version
 
-Kubernetes **v1.21** or higher is supported.
+Kubernetes **v1.22.17** or higher is supported.
```
Comment (travisn) [inline Documentation/Getting-Started/Prerequisites/prerequisites.md:10]: Let's just mention the minor version, not the patch version
```suggestion
Kubernetes **v1.22** or higher is supported.
```

File: Documentation/Getting-Started/quickstart.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ This guide will walk through the basic setup of a Ceph cluster and enable K8s ap
 
 ## Minimum Version
 
-Kubernetes **v1.21** or higher is supported by Rook.
+Kubernetes **v1.22.17** or higher is supported by Rook.
```
Comment (travisn) [inline Documentation/Getting-Started/quickstart.md:15]:
```suggestion
Kubernetes **v1.22** or higher is supported by Rook.
```

File: Documentation/Upgrade/rook-upgrade.md [hunk 5]
```diff
@@ -48,7 +48,7 @@ those releases.
 
 ## Breaking changes in v1.12
 
-* The minimum supported version of Kubernetes is v1.21.0.
+* The minimum supported version of Kubernetes is v1.22.17.
```
Comment (travisn) [inline Documentation/Upgrade/rook-upgrade.md:51]:
```suggestion
* The minimum supported version of Kubernetes is v1.22
```

File: Documentation/Getting-Started/Prerequisites/prerequisites.md [hunk 6]
```diff
@@ -7,7 +7,7 @@ and Rook is granted the required privileges (see below for more information).
 
 ## Minimum Version
```
Comment (travisn) [inline Documentation/Getting-Started/Prerequisites/prerequisites.md:8]: The helm docs also need to be updated, looks like the previous version they weren't updated. Please search for `Kubernetes 1.19+` and update those as well.

**Review by travisn — CHANGES_REQUESTED** [review]
A few more small changes, thanks!

File: Documentation/Getting-Started/Prerequisites/prerequisites.md (hunk 6 above)
Comment (Sheetalpamecha) [reply Documentation/Getting-Started/Prerequisites/prerequisites.md:8]: Sure Done

File: Documentation/Upgrade/rook-upgrade.md (hunk 5 above)
Comment (Sheetalpamecha) [reply Documentation/Upgrade/rook-upgrade.md]: Done

File: Documentation/Getting-Started/quickstart.md (hunk 4 above)
Comment (Sheetalpamecha) [reply Documentation/Getting-Started/quickstart.md]: Done

File: Documentation/Getting-Started/Prerequisites/prerequisites.md (hunk 3 above)
Comment (Sheetalpamecha) [reply Documentation/Getting-Started/Prerequisites/prerequisites.md]: Done

**Review by travisn — APPROVED** [review]

**Pull request merged**